lib/pure/asyncfile.nim

```diff
@@ -454,8 +454,15 @@ proc write*(f: AsyncFile, data: string): Future[void] =
 
     proc cb(fd: AsyncFD): bool =
       result = true
-      let remainderSize = data.len-written
-      let res = write(fd.cint, addr copy[written], remainderSize.cint)
+
+      let remainderSize = data.len - written
+
+      let res =
+        if data.len == 0:
+          write(fd.cint, copy.cstring, 0)
```

Comment:
what is a write of size zero supposed to do?

Reply:
When the written string is empty, create a new file and write nothing. Before it will crash because empty string can't be indexed.

```nim
addr copy[written]
```

Reply:
Sure but why not omit the ``write`` call then?

Reply:
Because POSIX in all of their wisdom allows `write()` with size `0` to perform actions such as error checking.